src/Interpreters/ArrayJoinAction.h

```diff
@@ -11,6 +11,13 @@ namespace DB
 class IFunctionOverloadResolver;
 using FunctionOverloadResolverPtr = std::shared_ptr<IFunctionOverloadResolver>;
 
+class DataTypeArray;
+class ColumnArray;
+
+const DataTypeArray * getArrayJoinDataType(const DataTypePtr & type, bool allow_map = true);
+const ColumnArray * getArrayJoinColumn(const ColumnPtr & column, bool allow_map = true);
```

Comment:
Is it used somewhere with `allow_map = false`?

Reply:
No. I keep this in case settings like allow_map is need. I'll remove it if think it is ok.